python/sqlalchemy/src/data_access_util.py

```diff
@@ -0,0 +1,143 @@
+import logging
+from sqlalchemy.orm import sessionmaker, relationship
+from model import User, Order, OrderLine, Product
+import config as cfg
+from sqlalchemy import create_engine
+import traceback
+
+
+class DataAccessUtil:
+
+    def __init__(self):
+        engine = create_engine('postgresql://%s:%s@%s:%s/%s' % 
+                            (cfg.db_user, cfg.db_password, cfg.db_host, cfg.db_port, cfg.database), echo=True)
+        self.Session = sessionmaker(bind=engine)
+
+    def get_users(self):
+        session = self.Session()
+
+        try:
+            users = session.query(User).all()
+            users_json = {
+                'users': [
+                    user.to_json() for user in users
+                ]
+            }
+            session.commit()
+            return users_json
+        except Exception as e:     
+            logging.debug('*** Exception in get_users: %s' % e)
+            session.rollback()
+            raise
+        finally:
+            session.close()
+
+    def list_orders_for_user(self, rq_user_id):
+        session = self.Session()
+
+        try:
+            user = session.query(User).get(rq_user_id)
+
+            if user is None:
+                logging.debug('Unable to find user with id = %s' % rq_user_id)
+                # TODO: Raise an error
+
+            user_json = user.to_json()
+
+            user_json['orders'] = [
+                order.to_json() for order in user.orders
+            ]
+            return user_json
+        except Exception as e:     
+            logging.debug('*** Exception in list_orders_for_user: %s' % e)
+            session.rollback()
+            raise
+        finally:
+            session.close()
+
+    def create_order(self, rq_user_id, products):
+        session = self.Session()
+
+        try:
+            user = session.query(User).get(rq_user_id)
+
+            if user is None:
+                raise Exception('Unable to find user with id = %s' % rq_user_id)
+
+            order_line_list = []
+            user_order = Order(user_id=rq_user_id, order_total=0)
+
+            # logging.debug('Processing %s products' % len(products))
```

Comment:
Drop commented out line, or uncomment this debug.

Reply:
Done